projects/plugins/jetpack/tools/check-block-assets.php

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env php
+<?php
+/**
+ * Script to check blocks' view.asset.php for unexpected dependencies.
+ *
+ * @package automattic/jetpack
+ */
+
+// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped -- This is not WordPress code.
+
+/**
+ * Non-WP dependencies to check for.
+ *
+ * @var string[]
+ */
+$bad_deps = array(
+	'jquery',
+	'lodash',
+	'lodash-es',
+	'moment',
+	'react',
+	'react-dom',
+);
+
+/**
+ * WP dependencies to allow.
+ *
+ * @var string[]
+ */
+$ok_wp_deps = array(
+	'wp-a11y',
+	'wp-api-fetch',
+	'wp-autop',
+	'wp-blob',
+	'wp-block-serialization-default-parser',
+	'wp-block-serialization-spec-parser',
+	'wp-deprecated',
+	'wp-dom',
+	'wp-dom-ready',
+	'wp-escape-html',
+	'wp-experiments',
+	'wp-hooks',
+	'wp-html-entities',
+	'wp-i18n',
+	'wp-is-shallow-equal',
+	'wp-polyfill',
+	'wp-preferences-persistence',
+	'wp-priority-queue',
+	'wp-redux-routine',
+	'wp-shortcode',
+	'wp-token-list',
+	'wp-url',
+	'wp-warning',
+);
+
+/**
+ * Blocks allowed to have extra dependencies.
+ *
+ * @var string[][] Keys are block names, value is an array of dependencies to not complain about.
+ */
+$allowed = array(
```

Comment:
These are all the current blocks with dependencies the script would complain about. Feel free to consider this a todo list. 😀